File: ccif/src/java/other/org/himalay/ccif/NetAddressReport.java
// Copyright (2013) Krishna C Tripathi. All rights reserved.
// 
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
package org.himalay.ccif ;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import java.lang.Exception;
import java.io.ByteArrayInputStream;
import java.io.PrintStream;
import org.himalay.msgs.runtime.Created;

import org.himalay.msgs.runtime.*;
@Created(date = "Fri Jan 15 01:33:04 EST 2016")

public  class NetAddressReport extends BinStruct implements PublicBinMsg{




    // members variables
    // ntAddrs
    public ArrayList<NetAddressReportMsg> ntAddrs ;

        public NetAddressReport () // throws Exception
    {
            init();
    }
    
    private void init()
    {
        // Initialize ntAddrs
        ntAddrs= new ArrayList<NetAddressReportMsg>();ntAddrs.setMemberSize(0);
    }

    public int readNoHeader(DataInputStream istream) throws IOException 
    {
    
     return read(istream);
       }
   
    public int read(DataInputStream istream) throws IOException 
    {
    preRead();
        int retVal= 0;

                     // read ntAddrs
        for (; istream.available() > 0 ; ){     NetAddressReportMsg temp;    temp = new NetAddressReportMsg();    retVal += temp.read(istream);    ntAddrs.add(temp);}

        postRead();
        return retVal;
    }


    public int write(DataOutputStream ostream) throws IOException 
    {
    preWrite();
        int retVal= 0;

        { /** fix dependent sizes for ntAddrs**/  }
    
                // write ntAddrs
        {ArrayList<NetAddressReportMsg> temp1 = ntAddrs;for (int iIdx=0; iIdx < temp1.getCount() ; iIdx++){     NetAddressReportMsg temp2    =    temp1.get(iIdx);    if (temp2!=null)retVal +=temp2.write(ostream);}}
postWrite();
        return retVal;
    }
    
    public int dump(DumpContext dc) throws IOException 
    {
        dc.indent();dc.getPs().print("NetAddressReport\n");
    dc.increaseIndent();
        int retVal= 0;
        // write ntAddrs
        { ArrayList<NetAddressReportMsg> temp1 = ntAddrs;for (int iIdx=0; iIdx < temp1.getCount() ; iIdx++){     NetAddressReportMsg element    = temp1.get(iIdx); dc.indent(); dc.getPs().println(iIdx);    if ( element != null ) {dc.indent();dc.getPs().println("element") ;retVal +=element.dump(dc);}}}
dc.decreaseIndent();
        return retVal;
    }


        // Getter for ntAddrs
    //public ArrayList<NetAddressReportMsg> getNtAddrs()
    //{
    //    return ntAddrs ;
    //}

    
    // Setter for ntAddrs
    //public void setNtAddrs(ArrayList<NetAddressReportMsg> val)
    //{
    //    this.ntAddrs= val;
    //}


    public int addToNtAddrs(NetAddressReportMsg val)
    {
            ntAddrs.add(val); return ntAddrs.size();
    }
    
    
    public int removeFromNtAddrs(NetAddressReportMsg val)
    {
            ntAddrs.remove(val); return ntAddrs.size();
    }
    
    
    public int removeNthFromNtAddrs(int idx)
    {
            ntAddrs.remove(idx); return ntAddrs.size();
    }
    
    
    public int emptyNtAddrs(int idx)
    {
            ntAddrs.clear(); return ntAddrs.size();
    }
    
    
    public int getSize() throws IOException
    {
       DataOutputStream dos= new DataOutputStream(new NullStream());
       return this.write(dos);
    }
    
        

}

// End of code

File: libccif/src/bp/NetAddressReport.cpp
//**//
//@Created(date = "Mon Feb 01 22:30:16 EST 2016")
// Copyright (2012) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
//**//
#include "stdio.h"
#include "NetAddressReport.h"

#ifdef org_himalay_ccif_NetAddressReport__USE_SMART_PTR
#define ArrayList(x) SmartPtrList<x>
#define ArrayList_iterator(x) SmartPtrList<x>::iterator
#include "SmartPtrList.h"
#else
#define ArrayList(x) BinMessagePtrList
#define ArrayList_iterator(x) BinMessagePtrList::iterator
#include "BinMessagePtrList.h"
#endif

// Namespace
namespace org {
namespace himalay {
namespace ccif {

using namespace ::org::himalay::msgs::runtime;

// Constructor
NetAddressReport::NetAddressReport() {
    references= 0;


    // ntAddrs
    ntAddrs.setMemberSize(0);
}
// Destructor
NetAddressReport::~NetAddressReport() {
    // ntAddrs
    {
#ifndef org_himalay_ccif_NetAddressReportMsg_USE_SMART_PTR
        ntAddrs.deleteMembers();
#endif
    }
}

int NetAddressReport::readNoHeader(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;

    // ntAddrs
    for (; istream.available() > 0 ; ) {
        org_himalay_ccif_NetAddressReportMsg__PTR_TYPE temp;
        org_himalay_ccif_NetAddressReportMsg__NEW( temp);
        retVal += temp->read(istream);
        ntAddrs.add(temp);
    }
    return retVal;
}

int NetAddressReport::read(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;
    // read ntAddrs
    for (; istream.available() > 0 ; ) {
        org_himalay_ccif_NetAddressReportMsg__PTR_TYPE temp;
        org_himalay_ccif_NetAddressReportMsg__NEW( temp);
        retVal += temp->read(istream);
        ntAddrs.add(temp);
    }

    return retVal;
}

int NetAddressReport::write(DataOutputStream& ostream) { // throws IOException
    {   /** fix dependent sizes for ntAddrs**/
    }

    int retVal= 0;
    // write ntAddrs
{   for (ArrayList_iterator(NetAddressReportMsg) it=  ntAddrs.getIterator() ; it.hasNext();) {
            {
                retVal +=it.get()->write(ostream);
            }
        }
    }

    return retVal;
}

int NetAddressReport::dump(DumpContext& dc) { // throws IOException
    dc.indent();
    dc.getPs().print("NetAddressReport\n");
    dc.increaseIndent();
    int retVal= 0;
    // write ntAddrs
{   for (ArrayList_iterator(NetAddressReportMsg) it=  ntAddrs.getIterator() ; it.hasNext();) {
            {
                dc.indent();
                dc.getPs().println("element");
                retVal +=it.get()->dump(dc);
            }
        }
    }
    dc.decreaseIndent();
    return retVal;
}


// Getter for ntAddrs
//ArrayList(NetAddressReportMsg) NetAddressReport::getNtAddrs()
//{
//return this->ntAddrs;
//}

// Setter for ntAddrs
//void NetAddressReport::setNtAddrs(ArrayList(NetAddressReportMsg) val)
//{
//this->ntAddrs= val;
//}

int NetAddressReport::getSize() { // throws IOException
    int retVal = 0;

    return retVal;
}

#ifdef org_himalay_ccif_NetAddressReport__USE_SMART_PTR
void intrusive_ptr_add_ref(org::himalay::ccif::NetAddressReport* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::ccif::NetAddressReport* p) {
    p->decreaseRef();
};;
#endif


}
}
}

// End of code